storm-core/src/clj/backtype/storm/daemon/supervisor.clj

```diff
@@ -454,10 +455,82 @@
       (shutdown-worker supervisor id))
     ))
 
+(defn get-blob-localname
+  "Given the blob information either gets the localname field if it exists,
+  else routines the default value passed in."
+  [blob-info defaultValue]
+  (if-let [val (if blob-info (get blob-info "localname") nil)] val defaultValue))
+
+(defn should-uncompress-blob?
+  "Given the blob information returns the value of the uncompress field, handling it either being
+  a string or a boolean value, or ifs its not specified then returns false"
+  [blob-info]
+  (boolean (and blob-info
+             (if-let [val (get blob-info "uncompress")]
+               (.booleanValue (Boolean. val))))))
```

Comment:
```Clojure
(Boolean. (get blob-info "uncompress"))
```
This should work for the cases we care about:
```
user=> (defn b [m k] (Boolean. (get m k)))
#'user/b
user=> (b nil "uncompress")
false
user=> (b {} "uncompress")
false
user=> (b {"uncompress" nil} "uncompress")
false
user=> (b {"uncompress" "false"} "uncompress")
false
user=> (b {"uncompress" "true"} "uncompress")
true
user=> (b {"uncompress" "TRUE"} "uncompress")
true
```

Also, in docstring `ifs its -> if it's`